java/runtime/src/main/java/io/ray/runtime/object/ObjectStore.java

```diff
@@ -72,12 +76,16 @@ public ObjectId put(Object object) {
    * @param ownerActorId The id of the actor to assign ownership.
    * @return Id of the object.
    */
-  public ObjectId put(Object object, ActorId ownerActorId) {
+  public ObjectId put(Object object, ActorId ownerActorId, Boolean isArrowData) {
```

Comment:
If we indeed have to introduce a flag, I prefer an enum. e.g. `DataType::ARROW` and `DataType::JavaObject`, so we can support `DataType::FLATBUFFER` in the future.

Reply:
Agreed, I think it should be a global enum, and will error if the metadata specifies a ser/deser enum that is not supported by that language.